massgov/mayflower-patternlab
# DP-5234/DP-5177 Refine and Refactor Patterns

## Description
Several patterns have been updated to make them more modular by including other patterns, more flexible by adding options, or more maintainable by using proper variables, the documentation has been updated to use ` instead of ' for variable names, and unused Patterns were tagged as obsolete.

## Related Issue / Ticket
https://jira.state.ma.us/browse/DP-5234

## Steps to Test
1. Updated NPM -> Remove your styleguide/node_modules folder, then run npm install and gulp to verify that gulp still runs without errors.

#### Impacted Areas in Application
* The package.json file was upgraded to work with npm 5.2.0
* Button - removed extra white space around text
* Input checkbox - Added optional icon (used on location filters)
* Page Title - new atom (h1)
* Cat Icon - switched `small` variable to type boolean
* Action Step - replaced download links with Form Downloads organism
* Arrow Nav - replaced `a` with link atom
* Back Button - refactored to use Button atom
* Banner Credit - new molecule.
* Event Filters - converted submit button to a button atom
* Icon Links - added Comp Heading
* Image Promo - refactored data object to make description == richText
* Location Filters - removed Checkbox Icon (use new optional icon with checkbox atom instead)
* Page Intro - new molecule
* Search Banner Form - changed the data object to use the proper namespace
* Callout Alert - refactored data object to make calloutAlert == decorativeLink (doesn't included that pattern anymore)
* Callout Time - Added option to change the icon the used.
* Form Requirements - changed to use an `h2` for the title.
* Helpful Links - Marked as obsolete - replaced with linked list
* Link List - Added option to hide bullets
* Rich Text - removed property tag
* Search Banner - Updated markup and data object to include other patterns
* Stacked Row Section - Marked as obsolete
* Suggested Pages - Refactored to use an image atom
* Emergency Alerts - Changed code to allow the header to be rendered by itself.
* Transition Page - replaced submit button with button atom
* Services Template - updated to remove sidebar heading above link list.
* Stacked Row Template - updated to contain code from the stacked row section organism
* Error Pages - updated to use Link List instead of Helpful Links
* Homepage - moved the Search Banner into the pre-content area.

#### Notes
* This PR includes code from another PR -> https://github.com/massgov/mayflower/pull/547

#### Today I learned...
